libsolutil/Numeric.h

```diff
@@ -29,7 +29,15 @@
 #error "Unsupported Boost version. At least 1.65 required."
 #endif
 
+// TODO: do this only conditionally as soon as a boost version with gcc 12 support is released.
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
```

Comment:
This is rather ugly... I'd have expected the build by default to consider boost headers as system libraries and not to have ``-Werror`` apply to them - not entirely sure why that's not the case...